### PR TITLE
fix(Script/ZulGurub) - High Priestess Jeklik improvements

### LINKED ISSUE #11380: [Zul'Gurub] High priestess Jeklik - High frequency for Sonic burst
### Current Behaviour

The cast the spell "Sonic burst" (a silence spell) every 10sec.

### Expected Blizzlike Behaviour

Jeklik should cast "Sonic burst" between 20sec and 30/40sec (first video timer is 30sec/40sec/30sec, second video: first cast 17sec, 2nd cast at 37sec)

### Source

First video - https://youtu.be/kJSj8MTAHo8?t=50
Second video - https://youtu.be/2nshuCKVk7o?t=17 
### Steps to reproduce the problem

Engage Jeklik and look the frequency with sonic burst.

### Extra Notes

_No response_

### AC rev. hash/commit

e13f427

### Operating system

Windows 11

### Custom changes or Modules

_No response_

### LINKED ISSUE #11381: [Zul'Gurub] High priestess Jeklik - Missing spells
### Current Behaviour

Actually, the boss cast Sonic burst and terrifying screech in P1.

Not cast Curse of Blood in P2.

WoWScrnShot_041222_175724

### Expected Blizzlike Behaviour

She should cast Sonic burst, Swoop, Pierce armor and blood leech in P1 and Curse of Blood in P2.

To Do:

- Adds Swoop (https://classic.wowhead.com/spell=18144/swoop), Pierce armor (https://classic.wowhead.com/spell=6016/pierce-armor), and blood leech (https://classic.wowhead.com/spell=22644/blood-leech)

- Add Curse of blood in P2 (https://www.wowhead.com/spell=16098/curse-of-blood)

### Source

https://wowpedia.fandom.com/wiki/High_Priestess_Jeklik_(tactics)

https://youtu.be/ZJoKmit_ess?t=1

### Steps to reproduce the problem

Engage Jeklik and look logs chat in everythings

### Extra Notes

IDs from Wowhead are maybe not good because spells seems duplicate wich differents values

_No response_

### AC rev. hash/commit

e13f427

### Operating system

Windows 11

### Custom changes or Modules

_No response_

### LINKED ISSUE #11382: [Zul'Gurub] High priestess Jeklik - Fly movements in P2
### Current Behaviour

She levitates and she moves as flying/swimming

<details>

WoWScrnShot_041222_165730

WoWScrnShot_041222_165710

</details>

### Expected Blizzlike Behaviour

She is in humanoid form, so she should walk and not levitate.

### Source

https://youtu.be/ZJoKmit_ess?t=49

https://youtu.be/nRuGGCQ5hcs?t=340

### Steps to reproduce the problem

Engage Jeklik and go in P2

### Extra Notes

She seems ignore "SetCanFly(false)" when she pass in P2.

### AC rev. hash/commit

e13f427

### Operating system

Windows 11

### Custom changes or Modules

_No response_

### LINKED ISSUE #11390: [Zul'Gurub] High Priestess Jeklik - Psychic scream
### Current Behaviour

She don't cast Psychic scream but Terrifying scream and only in P1 (an AoE fear)

### Expected Blizzlike Behaviour

Change the wrong spell ID: "Psychic scream" (https://www.wowhead.com/spell=22884/psychic-scream)

<s>The frequency is a little obscure. This spell seems to be cast during P1 and P2, but come very rarely (probably because fights are too short). Unable to find a good source about the delay.</s> According to the DBM timer on several videos, it's seems that this spells is cast between 30sec and 1min (I have see this spell only cast one time per fight and generally, never cast).

EDIT 04/20: After severals research he seems that this spell is only cast in P2.

### Source

https://youtu.be/T9_oghoERTc?

https://youtu.be/nRuGGCQ5hcs?

https://youtu.be/ZJoKmit_ess?

 https://youtu.be/kJSj8MTAHo8?t=50

https://youtu.be/63t9EXBfc44?t=569

Several videos here: http://progression-annexe.blogspot.com/2016/02/videos-of-retail-sources-for-vanilla_58.html

### Steps to reproduce the problem

Engage Jeklik

### Extra Notes

_No response_

### AC rev. hash/commit

5bebb14

### Operating system

Windows 11

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by Maelthyrr**
## Changes Proposed:
- Add missing spells
- Fix fly issue in P2
- Fix Chain Mind Flay
- Fix Psychic Scream issue
- Adjust timers spells
- Add missing emotes
- Add missing aura before the fight 

## Issues Addressed:

- Closes #11381
- Closes #11382
- Closes a part of #11386
- Closes #11390 
- Closes #11380

## SOURCE:

Missing emotes, flying fix in P2, and few timers pick from https://github.com/TrinityCore/TrinityCore/pull/27274
(Thanks offl for you PR)

## Extra Notes:
After severals researches (TC, MangosClassic, other projects and issues...). It seems that nobody have found the right spell IDs from 
Chain mind flee. Also, this spells is rarely mentionned (maybe not exist on retail). So I deleted this spell for just let "Mind flay"

## Tests Performed:

- Tested IG

## How to Test the Changes:

1. .tele zulgurub
2. Engage Jeklik
3. Check that all spells are cast
4. Check some potential issues (except with bomb in P2, know bug)

## Known Issues and TODO List:

- [X] Timers need adjustements

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
You are assigning creature scripts to creatures but you havent scripted them in the core. Therefore the errors

**Comment by Lynethris** [comment]
> You are assigning creature scripts to creatures but you havent scripted them in the core. Therefore the errors

Yes, I remove this modifs. I thought that simple properties modifications with bats in P1, but it seems to be related to flying bats in P2 (not touched in this PR).

**Comment by temperrr** [comment]
- [x] She should cast Sonic burst, Swoop, Pierce armor and blood leech in P1 and Curse of Blood in P2.
- [ ] She isflying in humanoid form during p1, she should be a bat in p1
- [x] Using physic scream in p2
- [x] using mind flay in p2

**Comment by Lynethris** [comment]
@temperrr I'm unable to reproduce this.

**Comment by temperrr** [comment]
Works now. Just one odd thing, if you wipe during p1 she flies back in humanoid form which looks odd.

**Comment by Lynethris** [comment]
> Works now. Just one odd thing, if you wipe during p1 she flies back in humanoid form which looks odd.

Yes, I notice this too. But not related to  this PR. I will reopen a new issue for this.

File: data/sql/updates/pending_db_world/rev_1650481776270749700.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1650481776270749700');
+
+-- Fix for the levitate issue
+UPDATE `creature_template` SET `speed_walk` = 1, `speed_run` = 2.14286, `flags_extra` = `flags_extra`|512 WHERE `entry` = 14517;
+UPDATE `creature_template_movement` SET `Flight` = 0 WHERE `CreatureId` = 14517;
+-- Update Emotes
+DELETE FROM `creature_text` WHERE `CreatureID` = 14517 AND `GroupID` IN (3,4);
+INSERT INTO `creature_text` (`CreatureID`,`GroupID`,`ID`,`Text`,`Type`,`Language`,`Probability`,`Emote`,`Duration`,`Sound`,`BroadcastTextId`,`TextRange`,`comment`) VALUES
+(14517,3,0,"%s emits a deafening shriek!",16,0,100,0,0,0,10370,0,"High Priestess Jeklik"),
+(14517,4,0,"%s begins to cast a Great Heal!",16,0,100,0,0,0,10494,0,"High Priestess Jeklik");
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1650481776270749700.sql:10]: newline

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 2]
```diff
@@ -22,20 +22,31 @@
 enum Says
 {
     SAY_AGGRO                   = 0,
-    SAY_RAIN_FIRE               = 1,
-    SAY_DEATH                   = 2
+    SAY_CALL_RIDERS             = 1,
+    SAY_DEATH                   = 2,
+    EMOTE_SUMMON_BATS           = 3,
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:27]: you never use this in script

File: data/sql/updates/pending_db_world/rev_1650481776270749700.sql (hunk 1 above)
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1650481776270749700.sql:10]:
```suggestion
(14517,3,0,"%s emits a deafening shriek!",16,0,100,0,0,0,10370,0,"High Priestess Jeklik - EMOTE_SUMMON_BATS"),
(14517,4,0,"%s begins to cast a Great Heal!",16,0,100,0,0,0,10494,0,"High Priestess Jeklik - EMOTE_GREAT_HEAL");
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1650481776270749700.sql (hunk 1 above)
Comment (Lynethris) [reply data/sql/updates/pending_db_world/rev_1650481776270749700.sql]: fix

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp (hunk 2 above)
Comment (Lynethris) [reply src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:27]: Yes, it's fix now.

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 3]
```diff
@@ -142,53 +165,69 @@ class boss_jeklik : public CreatureScript
...
+                        DoCast(me, SPELL_CURSE_OF_BLOOD);
+                        events.ScheduleEvent(EVENT_CURSE_OF_BLOOD, urand(25000, 30000), PHASE_TWO);
+                        break;
+                    case EVENT_PSYCHIC_SCREAM:
+                        DoCastVictim(SPELL_PSYCHIC_SCREAM);
+                        events.ScheduleEvent(EVENT_PSYCHIC_SCREAM, urand(35000, 45000), PHASE_TWO);
                         break;
                     case EVENT_SHADOW_WORD_PAIN:
                         if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
                             DoCast(target, SPELL_SHADOW_WORD_PAIN);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:212]:
```suggestion
                        DoCastRandomTarget((SPELL_SHADOW_WORD_PAIN, 0, true);
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 4]
```diff
@@ -142,53 +165,69 @@ class boss_jeklik : public CreatureScript
...
                         if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
                             for (uint8 i = 0; i < 6; ++i)
                                 if (Creature* bat = me->SummonCreature(NPC_BLOODSEEKER_BAT, SpawnBat[i], TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
                                     bat->AI()->AttackStart(target);
-                        events.ScheduleEvent(EVENT_SPAWN_BATS, 60000, 0, PHASE_ONE);
+                        events.ScheduleEvent(EVENT_SPAWN_BATS, 30000, PHASE_ONE);
+                        break;
+                    //Phase two
+                    case EVENT_CURSE_OF_BLOOD:
+                        DoCast(me, SPELL_CURSE_OF_BLOOD);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:203]:
```suggestion
                        DoCastSelf(SPELL_CURSE_OF_BLOOD);
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 5]
```diff
@@ -142,53 +165,69 @@ class boss_jeklik : public CreatureScript
...
-                        DoCastVictim(SPELL_CHAIN_MIND_FLAY);
-                        events.ScheduleEvent(EVENT_CHAIN_MIND_FLAY, urand(15000, 30000), 0, PHASE_TWO);
+                        events.ScheduleEvent(EVENT_MIND_FLAY, urand(20000, 40000), PHASE_TWO);
                         break;
                     case EVENT_GREATER_HEAL:
+                        Talk(EMOTE_GREAT_HEAL);
                         me->InterruptNonMeleeSpells(false);
-                        DoCast(me, SPELL_GREATERHEAL);
-                        events.ScheduleEvent(EVENT_GREATER_HEAL, urand(25000, 35000), 0, PHASE_TWO);
+                        DoCast(me, SPELL_GREATER_HEAL);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:222]:
```suggestion
                        DoCastSelf(SPELL_GREATER_HEAL);
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 6]
```diff
@@ -100,30 +118,35 @@ class boss_jeklik : public CreatureScript
         {
             _EnterCombat();
             Talk(SAY_AGGRO);
+            me->RemoveAurasDueToSpell(SPELL_GREEN_CHANNELING);
+            me->SetDisableGravity(true);
+            DoCast(me, SPELL_BAT_FORM);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:123]:
```suggestion
            DoCastSelf(SPELL_BAT_FORM);
```

File: data/sql/updates/pending_db_world/rev_1650481776270749700.sql [hunk 7]
```diff
@@ -0,0 +1,10 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1650481776270749700');
+
+-- Fix for the levitate issue
+UPDATE `creature_template` SET `speed_walk` = 1, `speed_run` = 2.14286, `flags_extra` = `flags_extra`|512 WHERE `entry` = 14517;
+UPDATE `creature_template_movement` SET `Flight` = 0 WHERE `CreatureId` = 14517;
+-- Update Emotes
+DELETE FROM `creature_text` WHERE `CreatureID` = 14517 AND `GroupID` IN (3,4);
+INSERT INTO `creature_text` (`CreatureID`,`GroupID`,`ID`,`Text`,`Type`,`Language`,`Probability`,`Emote`,`Duration`,`Sound`,`BroadcastTextId`,`TextRange`,`comment`) VALUES
+(14517,3,0,"%s emits a deafening shriek!",16,0,100,0,0,0,10370,0,"High Priestess Jeklik - EMOTE_SUMMON_BATS"),
+(14517,4,0,"%s begins to cast a Great Heal!",16,0,100,0,0,0,10494,0,"High Priestess Jeklik - EMOTE_GREAT_HEAL");
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1650481776270749700.sql:10]:
```suggestion
(14517,3,0,'%s emits a deafening shriek!',16,0,100,0,0,0,10370,0,'High Priestess Jeklik - EMOTE_SUMMON_BATS'),
(14517,4,0,'%s begins to cast a Great Heal!',16,0,100,0,0,0,10494,0,'High Priestess Jeklik - EMOTE_GREAT_HEAL');
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 8]
```diff
@@ -87,6 +104,7 @@ class boss_jeklik : public CreatureScript
 
         void Reset() override
         {
+            DoCast(me, SPELL_GREEN_CHANNELING);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:107]:
```suggestion
            DoCastSelf(SPELL_GREEN_CHANNELING);
```

**Review by Nefertumm — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1650481776270749700.sql [hunk 9]
```diff
@@ -0,0 +1,10 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1650481776270749700');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1650481776270749700.sql:1]: Remove this line, is not useful anymore.

**Review by Nefertumm — APPROVED** [review]

**Review by Kitzunu — APPROVED** [review]

**Comment by Nefertumm** [comment]
Update your branch with current master so the sql can pass the check. 👍

**Pull request merged**